src/MoBi.Presentation/UICommand/ExportBuildingBlockToExcelUICommand.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections.Generic;
+using System.Data;
+using MoBi.Assets;
+using MoBi.Core.Services;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Services;
+using OSPSuite.Infrastructure.Export;
+using OSPSuite.Presentation.UICommands;
+using OSPSuite.Utility;
+using OSPSuite.Utility.Extensions;
+
+namespace MoBi.Presentation.UICommand
+{
+   public abstract class ExportBuildingBlockToExcelUICommand<T> : ObjectUICommand<T> where T : class, IBuildingBlock
+   {
+      private readonly IMoBiProjectRetriever _projectRetriever;
+      private readonly IDialogCreator _dialogCreator;
+      private readonly IMapper<T, List<DataTable>> _mapper;
+
+      protected ExportBuildingBlockToExcelUICommand(
+         IMoBiProjectRetriever projectRetriever,
+         IDialogCreator dialogCreator,
+         IMapper<T, List<DataTable>> mapper)
+      {
+         _projectRetriever = projectRetriever;
+         _dialogCreator = dialogCreator;
+         _mapper = mapper;
+      }
+
+      protected override void PerformExecute()
+      {
+         var currentProject = _projectRetriever.CurrentProject;
+         var projectName = currentProject.Name;
+         if (string.IsNullOrEmpty(projectName))
+            projectName = AppConstants.Undefined;
+
+         var defaultFileName = AppConstants.DefaultFileNameForBuildingBlockExport(projectName, Subject.Name);
+         var excelFileName = _dialogCreator.AskForFileToSave(AppConstants.Captions.ExportToExcel, Constants.Filter.EXCEL_SAVE_FILE_FILTER, Constants.DirectoryKey.MODEL_PART, defaultFileName);
+
+         if (excelFileName.IsNullOrEmpty())
+            return;
+
+         var mappedValues = _mapper.MapFrom(Subject);
+         ExportToExcelTask.ExportDataTablesToExcel(mappedValues, excelFileName, openExcel: true);
```

Comment:
Would you please set a quick example on how this hierarchy should be exactly?
Since I can see some examples on the code but none reflects what I need, Meaning...
This class for example;:
InteractionTasksForPathAndValueEntity<TParent, TBuildingBlock, TBuilder>
Requires 3 Generic arguments, while this:
IInteractionTasksForExtendablePathAndValueEntity
has these 2:
  where TBuildingBlock : class, IBuildingBlock<TPathAndValueEntity>
  where TPathAndValueEntity : PathAndValueEntity

So not sure exactly how to make it fix without getting it wrong.

Reply:
I would put as much as possible in `InteractionTasksForPathAndValueEntity<TParent, TBuildingBlock, TBuilder>`.

Reply:
I`ve done it with everything BUT the mapper,since I`m having the problem that the mappers are of this types:
 IMapper<ParameterValuesBuildingBlock, List<DataTable>>
 IMapper<InitialConditionsBuildingBlock, List<DataTable>>
 IMapper<ExpressionProfileBuildingBlock, List<DataTable>>
 
 So the ExpressionProfile is different than the other two ( not an PathAndValueEntityBuildingBlock).
 So I think solving the mapper on the command is cleaner, would you agree?